br/pkg/metautil/metafile.go

```diff
@@ -51,24 +55,37 @@ const (
 	MetaV2
 )
 
-func Decrypt(content []byte, cipher *backuppb.CipherInfo) ([]byte, error) {
-	if encryptionpb.EncryptionMethod_UNKNOWN == cipher.CipherType {
-		return content, errors.Annotate(berrors.ErrInvalidArgument, "cipher type invalid")
-	} else if encryptionpb.EncryptionMethod_PLAINTEXT == cipher.CipherType {
-		return content, nil
+func Encrypt(content []byte, cipher *backuppb.CipherInfo) (encryptedContent, iv []byte, err error) {
+	switch cipher.CipherType {
+	case encryptionpb.EncryptionMethod_PLAINTEXT:
+		return content, iv, nil
+	case encryptionpb.EncryptionMethod_AES128_CTR,
+		encryptionpb.EncryptionMethod_AES192_CTR,
+		encryptionpb.EncryptionMethod_AES256_CTR:
+		// generate radom iv for aes crypter
+		iv = make([]byte, CrypterIvLen)
+		_, err = rand.Read(iv)
+		if err != nil {
+			return content, iv, errors.Trace(err)
+		}
+		encryptedContent, err = encrypt.AESEncryptWithCBC(content, cipher.CipherKey, iv)
+		return
+	default:
+		return content, iv, errors.Annotate(berrors.ErrInvalidArgument, "cipher type invalid")
 	}
-
-	return encrypt.AESDecryptWithECB(content, cipher.CipherKey)
 }
 
-func Encrypt(content []byte, cipher *backuppb.CipherInfo) ([]byte, error) {
-	if encryptionpb.EncryptionMethod_UNKNOWN == cipher.CipherType {
-		return content, errors.Annotate(berrors.ErrInvalidArgument, "cipher type invalid")
-	} else if encryptionpb.EncryptionMethod_PLAINTEXT == cipher.CipherType {
+func Decrypt(content []byte, cipher *backuppb.CipherInfo, iv []byte) ([]byte, error) {
+	switch cipher.CipherType {
+	case encryptionpb.EncryptionMethod_PLAINTEXT:
 		return content, nil
+	case encryptionpb.EncryptionMethod_AES128_CTR,
+		encryptionpb.EncryptionMethod_AES192_CTR,
+		encryptionpb.EncryptionMethod_AES256_CTR:
+		return encrypt.AESDecryptWithCBC(content, cipher.CipherKey, iv)
```

Comment:
CBC is not CTR

Reply:
The Package of encrypt In tidb repo has API of ECB/CBC/OFB/CFB , but not CTR。 I will add CTR API right now.